tools/github/generate-runtimes-body.ts

```diff
@@ -24,6 +26,20 @@ function getRuntimeInfo(srtoolReportFolder: string, runtimeName: string) {
   };
 }
 
+// Srtool expects the pallet parachain_system to be at index 1. However, in the case of moonbase,
+// the pallet parachain_system is at index 6, so we have to recalculate the hash of the
+// authorizeUpgrade call in the case of moonbase by hand.
```

Comment:
So this is the main motivation for calculating by hand right? Honestly, I would prefer to make a one-time breaking change to alphanet and re-index the pallets. I think the damage would be small (possibly nobody would even notice) because it only affects people submitting Substrate extrinsics (not ethereum) and the only real tool for doing that is Polkadot js which dynamically detects the pallet indices..

Reply:
Yes it's possible but it requires more work because we have to manage a migration. @crystalin prefers that we calculate the hash by hand.

Reply:
What kind of migration would we need? Anyway, I won't block this, but it seems like more work in the long run. Plus re-indexing moonbase would make it more similar to the other runtimes.

Reply:
It would be nice if we knew exactly what a change in pallet indexes would entail.
Does it change the prefix of the pallet for storage? If yes, then a migration is needed.

Reply:
It does not change anything about storage. It simply changes the way you construct your transaction. When you send a TX it is a scale-encoded `Call` enum (plus signed extra, plus signature). In scale enums can only have up to 255 variants so that they can be encoded in a single byte. `construct_runtime!` creates this "outer" `Call` enum for us with the given indices representing the corresponding variants. If we change the indices, then everyone constructing normal FRAME transactions will have to use the new indices. Since everyone who is doing that is using PolkadotJS which dynamically detects the indices from the metadata there is no risk.

Reply:
Ok you have convinced me, I am ok to change the order of the pallets instead.  @crystalin, do you agree?

Reply:
Wait... not so fast :p
What would happen to indexers like subscan and ...

Reply:
That's good question. I don't know how those indexers work. I'm probably out of my depth here.

Reply:
@mario-purestake or @albertov19 might have a clue on that or know who can answer that.

Reply:
For this case in particular we can calculate the hash by hand, it is not a problem.  But it would be important that we know exactly what a pallet reordering implies, because we may need to do it in the future!

Reply:
We did index the pallets in order to never have to reorder them (an index can be removed but not replaced). However things changes so we never know

Reply:
I can't be of much help on indexing the Substrate side or about Subscan, it's something I haven't taken a look :disappointed: I think Alberto can be more helpful regarding the contacts.

Reply:
We could probably ask Yakio (from Subscan) in Element, I've invited you @crystalin to the group (Joshy is in there as well).

Reply:
I asked, but anyway, we don't need to hold up this PR. If it turns out we can re-order them we can do it in a followup.

Reply:
![image](https://user-images.githubusercontent.com/2915325/134906289-551dda2b-5686-4db5-8d65-9629d3ba7c35.png)
Subscan has confirmed that re-indexing pallets will not cause problems for them :partying_face:

Reply:
So, i propose to replace this PR by #859